### ISSUE TITLE
group alignments by insertion

### ISSUE THREAD

**Issue by jrharting** [igvteam/igv#1121]
Is there a way to sort reads by an insertion?  I have some data where some alleles appear to differ only by a few 1-bp insertions and I'd like to see if they are in phase, but I am unable to sort/group by insertions at some position.

![image](https://user-images.githubusercontent.com/3595985/157761378-c76384fb-76a9-4407-9150-9b396cd98a4e.png)

**Comment by jrobinso** [comment]
Not currently.

**Comment by DarioS** [comment]
Ah, this answers [the question](https://groups.google.com/g/igv-help/c/ioSjJOWeBag/m/YmDnjU2cAwAJ) which I wrote a couple of days ago in igv-help forum.

**Comment by lbergelson** [comment]
@DarioS @jrharting It seems like there actually IS support for this to some degree.  If you select the expanded insertion position or the one before/after it and sort by `Sort Alignments By -> base` it should sort the insertions at that point by length.  

It currently doesn't take the content of the insertion into account, so if you have an AA insertion and a TT insertion it won't sort them separately.    It's a bit funky with adjacent insertions as well.  

I'm going to aim to improve it in the future but it requires a bit of thought or a rework of how sorting is handled so it's not a trivial fix.  

I think the existing sorting by length should be good enough in most cases though.

**Comment by jrobinso** [comment]
Yes I was incorrect, just to add to @lbergelson answer.   Sort is governed by the base/indel pileup at the center of the screen.  Unfortunately expanding insertions affects what is considered "center" in unpredictable ways.   This is a known issue.  So to be certain of sort position close expanded insertions, and center the view with the aid of the center line, enabled from the Alignments preference.

**Comment by jrobinso** [comment]
@jrharting Have to tried sorting with the insertions collapsed?  It should work, you can expand after the sort.
